src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java

```diff
@@ -330,7 +326,11 @@ public boolean isVersioned(File path) {
     }
 
     public List<File> listModifiedFiles(File path) throws SVNException {
-        SVNClientManager svnClientManager = SVNClientManager.newInstance();
+        SVNClientManager svnClientManager = clientManagerMap.get(path.getAbsolutePath());
+        if(svnClientManager == null) {
```

Comment:
These `svnClientManager` accesses should be guarded with a `synchronized` block in order to guarantee cross-thread visibility. Without synchronization the following is possible:

1. Thread A gets a `null` on line 330 and thus creates and inserts into the HashMap a new `SVNClientManager`.
2. Thread B comes along after A, and attempts to access the same `SVNClientManager` (aka from the same absolute path) but also receives a `null`. Thread B recreates the `SVNClientManager` and inserts it into the HashMap, probably leaving the one that A created 'leaked'.

Reply:
synchronized block added